src/transaction/atomic_replication_helper.cpp

```diff
@@ -125,26 +123,20 @@ namespace cublog
     PGBUF_INIT_WATCHER (&m_watcher, PGBUF_ORDERED_HEAP_NORMAL, PGBUF_ORDERED_NULL_HFID);
   }
 
-  atomic_replication_helper::atomic_replication_unit::~atomic_replication_unit ()
+  atomic_replication_helper::atomic_replication_unit::atomic_replication_unit (
+	  atomic_replication_helper::atomic_replication_unit &&that)
+    : m_record_lsa {std::move (that.m_record_lsa)}
+    , m_vpid {std::move (that.m_vpid)}
+    , m_record_index {std::move (that.m_record_index)}
   {
-    PGBUF_CLEAR_WATCHER (&m_watcher);
+    // ctor only used during construction; not allowed while operation is in progress
+    assert (m_page_ptr == nullptr);
+    assert (PGBUF_IS_CLEAN_WATCHER (&m_watcher));
   }
 
-  template <typename T>
-  void atomic_replication_helper::atomic_replication_unit::apply_log_redo (THREAD_ENTRY *thread_p,
-      log_rv_redo_context &redo_context, const log_rv_redo_rec_info<T> &record_info)
+  atomic_replication_helper::atomic_replication_unit::~atomic_replication_unit ()
   {
-    LOG_RCV rcv;
-    if (m_page_ptr != nullptr)
-      {
-	rcv.pgptr = m_page_ptr;
-      }
-    else
-      {
-	rcv.pgptr = m_watcher.pgptr;
-      }
-    rcv.reference_lsa = m_record_lsa;
-    log_rv_redo_record_sync_apply (thread_p,redo_context, record_info, m_vpid, rcv);
+    PGBUF_CLEAR_WATCHER (&m_watcher);
```

Comment:
IMHO
```suggestion
    assert (PGBUF_IS_CLEAN_WATCHER (&m_watcher));
```
although I am not so sure because a PTS can "go down" at any given time, therefore the dtor can occur at any given moment in the state of replication; better to add the assert here also with a comment that it might not hold; and we will see during testing